app/src/pages/inside/stepPage/modals/linkIssueModal/linkIssueModal.jsx

```diff
@@ -0,0 +1,263 @@
+import { GhostButton } from 'components/buttons/ghostButton';
+import { FormField } from 'components/fields/formField';
+import { Input } from 'components/inputs/input';
+import { FieldErrorHint } from 'components/fields/fieldErrorHint';
+import { validate, fetch } from 'common/utils';
+import styles from './linkIssueModal.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  linkButton: {
+    id: 'LinkIssueModal.linkButton',
+    defaultMessage: 'Link',
+  },
+  title: {
+    id: 'LinkIssueModal.title',
+    defaultMessage: 'Link issue',
+  },
+  linkSuccessMessage: {
+    id: 'LinkIssueModal.linkSuccessMessage',
+    defaultMessage: 'Completed successfully!',
+  },
+  addIssueIdTitle: {
+    id: 'LinkIssueModal.addIssueIdTitle',
+    defaultMessage: 'Add issue id',
+  },
+  addIssueButtonTitle: {
+    id: 'LinkIssueModal.addIssueButtonTitle',
+    defaultMessage: 'Add new issue',
+  },
+  issueLinkLabel: {
+    id: 'LinkIssueModal.issueLinkLabel',
+    defaultMessage: 'Link to issue',
+  },
+  issueIdLabel: {
+    id: 'LinkIssueModal.issueIdLabel',
+    defaultMessage: 'Issue ID',
+  },
+  issueLinkPlaceholder: {
+    id: 'LinkIssueModal.issueLinkPlaceholder',
+    defaultMessage: 'Enter link to issue',
+  },
+  closeModalWarning: {
+    id: 'LinkIssueModal.closeModalWarning',
+    defaultMessage: 'You have to save changes or cancel them before closing the window',
+  },
+  linkIssueSuccess: {
+    id: 'LinkIssueModal.linkIssueSuccess',
+    defaultMessage: 'Defect link is added',
+  },
+});
+
+@withModal('linkIssueModal')
+@injectIntl
+@reduxForm({
+  form: 'linkIssueForm',
+  validate: ({ issues }) => ({
+    issues:
+      issues &&
+      issues.map((issue) => ({
+        issueLink: (!issue.issueLink || !validate.url(issue.issueLink)) && 'urlHint',
+        issueId: (!issue.issueId || !validate.issueId(issue.issueId)) && 'issueIdHint',
+      })),
+  }),
+})
+@connect(
+  (state) => ({
+    url: URLS.testItemsAddIssues(activeProjectSelector(state)),
+    externalSystems: externalSystemSelector(state),
+  }),
+  {
+    showNotification,
+  },
+)
+export class LinkIssueModal extends Component {
+  static propTypes = {
+    intl: PropTypes.object.isRequired,
+    url: PropTypes.string.isRequired,
+    showNotification: PropTypes.func.isRequired,
+    externalSystems: PropTypes.array.isRequired,
+    initialize: PropTypes.func.isRequired,
+    handleSubmit: PropTypes.func.isRequired,
+    change: PropTypes.func.isRequired,
+    data: PropTypes.shape({
+      items: PropTypes.array,
+      fetchFunc: PropTypes.func,
+    }).isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.dropdownOptions = props.externalSystems.map((item) => ({
+      value: item.id,
+      label: item.project,
+    }));
+
+    this.props.initialize({
+      systemId: this.dropdownOptions[0].value,
+      issues: [{}],
+    });
+
+    this.state = {
+      isInputsHaveBeenChanged: false,
```

Comment:
That property is defined in ReduxForm API. So local state is unnecessary.
https://redux-form.com/7.4.2/docs/api/props.md/#-code-pristine-boolean-code-